craft_parts/utils/partition_utils.py

```diff
@@ -66,9 +67,6 @@ def validate_partition_names(partitions: Sequence[str] | None) -> None:
             "Partition feature is enabled but no partitions are defined."
         )
 
-    if partitions[0] != "default":
-        raise errors.FeatureError("First partition must be 'default'.")
```

Comment:
We can do a slightly different check here: the name `default` can only appear at index 0. You can't use a named default partition, and call something else `default`.